scipy/stats/_continuous_distns.py

```diff
@@ -3490,12 +3490,17 @@ def _rvs(self, mu, size=None, random_state=None):
         return random_state.wald(mu, 1.0, size=size)
 
     def _pdf(self, x, mu):
-        # invgauss.pdf(x, mu) =
-        #                  1 / sqrt(2*pi*x**3) * exp(-(x-mu)**2/(2*x*mu**2))
-        return 1.0/np.sqrt(2*np.pi*x**3.0)*np.exp(-1.0/(2*x)*((x-mu)/mu)**2)
+        return np.exp(self._logpdf(x, mu))
 
     def _logpdf(self, x, mu):
-        return -0.5*np.log(2*np.pi) - 1.5*np.log(x) - ((x-mu)/mu)**2/(2*x)
+        a = -0.5 * np.log(2 * np.pi) - 1.5 * np.log(x)
+        out = _lazywhere(
+            mu == np.inf,
+            (x, mu, a),
+            lambda x, mu, a: invgamma.logpdf(x, 0.5, scale=0.5),
+            f2=lambda x, mu, a: a - 0.5 * x / mu ** 2 + 1.0 / mu - 0.5 / x,
```

Comment:
This is a constant value so you don't need a lambda for it.

Reply:
Could you elaborate what you mean by it being a constant value? `f2` depends on `x` and `mu` so im not sure I understand what it means.